src/devices/Media/VideoDevice/Devices/UnixVideoDevice.cs

```diff
@@ -0,0 +1,546 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Iot.Device.Media
+{
+    /// <summary>
+    /// Represents a communications channel to a video device running on Unix.
+    /// </summary>
+    internal class UnixVideoDevice : VideoDevice
+    {
+        private const string DefaultDevicePath = "/dev/video";
+        private const int BufferCount = 4;
+        private int _deviceFileDescriptor = -1;
+        private static readonly object s_initializationLock = new object();
+
+        /// <summary>
+        /// Path to video resources located on the platform.
+        /// </summary>
+        public override string DevicePath { get; set; }
+
+        /// <summary>
+        /// The connection settings of the video device.
+        /// </summary>
+        public override VideoConnectionSettings Settings { get; }
+
+        /// <summary>
+        /// The max capture size of the video device.
+        /// </summary>
+        public override (uint Width, uint Height) MaxSize
+        {
+            get
+            {
+                v4l2_cropcap cropcap = new v4l2_cropcap()
+                {
+                    type = v4l2_buf_type.V4L2_BUF_TYPE_VIDEO_CAPTURE
+                };
+                V4l2Struct(VideoSettings.VIDIOC_CROPCAP, ref cropcap);
+
+                return (cropcap.bounds.width, cropcap.bounds.height);
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="UnixVideoDevice"/> class that will use the specified settings to communicate with the video device.
+        /// </summary>
+        /// <param name="settings">The connection settings of a video device.</param>
+        public UnixVideoDevice(VideoConnectionSettings settings)
+        {
+            Settings = settings;
+            DevicePath = DefaultDevicePath;
+
+            Initialize();
+        }
+
+        /// <summary>
+        /// Capture a picture from the video device.
+        /// </summary>
+        /// <param name="path">Picture save path</param>
+        public override async Task CaptureAsync(string path)
+        {
+            using (CancellationTokenSource cts = new CancellationTokenSource())
+            {
+                await CaptureAsync(path, cts.Token);
+            }
+        }
+
+        /// <summary>
+        /// Capture a picture from the video device.
+        /// </summary>
+        /// <param name="path">Picture save path</param>
+        /// <param name="token">A cancellation token that can be used to cancel the work</param>
+        public override async Task CaptureAsync(string path, CancellationToken token)
+        {
+            await Task.Run(() =>
```

Comment:
the token is mostly useless here - it won't automagically work so currently it will only have effect if you cancel before Task.Run gets called

Reply:
(suggestion is to use synchronous method here since async doesn't buy you anything)

Reply:
Actually, what I want is something like this😂. Also it can be cancelled from other threads
```C#
CancellationTokenSource cts = new CancellationTokenSource(1000);
await device.CaptureAsync($"{path}/jpg_direct_output.jpg", cts.Token);
```